### ISSUE TITLE
Update local notifications

### ISSUE THREAD

**Issue by ognjenns** [RestComm/restcomm-ios-sdk#668]
We should change the local notification and use APIs in the User Notifications framework instead
With it we can present alert, sound or increase badge while the app is in foreground too with iOS 10
We can handle all event in one place when user tapped (or slide) the action button, even while the app has already been killed.
Support 3D touch instead of sliding gesture.
Support Rich Notification with custom UI.

**Comment by atsakiridis** [comment]
@ognjenns makes sense. I added the `backgrounding` label to this, but please also add the same to any other issue you have opened for Push Notifications fixes/enhancements so that they are easy to find
